modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/MicroprofileServerCodegen.java

```diff
@@ -0,0 +1,74 @@
+package org.openapitools.codegen.languages;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+
+import org.openapitools.codegen.CodegenOperation;
+import org.openapitools.codegen.CodegenResponse;
+import org.openapitools.codegen.CodegenType;
+import org.openapitools.codegen.model.ModelMap;
+import org.openapitools.codegen.model.OperationMap;
+import org.openapitools.codegen.model.OperationsMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MicroprofileServerCodegen extends JavaClientCodegen {
```

Comment:
personally I would suggest NOT to extend a java server with JavaClientCodegen as most options available in JavaClientCodegen  won't make sense in the server generator if I understand it correctly.

Reply:
The java generator, using the microprofile library, generates an interface that could be used for either client or server.

This is what is being extended. I understand that it seems odd to have a server code generator inherit from a client generator. But we would have to re write the functionality of the java generator. 

Do you have a suggestion of how to proceed with this? 
Or perhaps it could be listed as a `MicroprofileClientCodegen` because it could be used for either client of server

Reply:
right seems odd but to me but it's not a showstopper.

if this way is a good starting point to add the server generator for microprofile, i'm totally ok with that.